javascript/0093-restore-ip-addresses.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @param {string} s
+ * @return {string[]}
+ */
+var restoreIpAddresses = function (s) {
+    let res = [];
+
+    if (s.length > 12) return res;
+
+    /**
+     *
+     * @param {number} i
+     * @param {number} dots
+     * @param {string} currentIP
+     */
+    function backtracking(i, dots, currentIP) {
+        if (dots === 4 && i == s.length) {
+            res.push(currentIP.slice(0, currentIP.length - 1));
+            return;
+        } else if (dots > 4) {
+            return;
+        }
+
+        for (let j = i; j < Math.min(i + 3, s.length); j++) {
+            if (+s.slice(i, j + 1) < 256 && (i == j || s[i] != '0')) {
+                backtracking(
+                    j + 1,
+                    dots + 1,
+                    currentIP + s.slice(i, j + 1) + '.'
+                );
+            }
+        }
+    }
+
+    backtracking(0, 0, '');
```

Comment:
I posted this in another submission and it applies here too. Btw thanks for doing multiple solutions!!

https://github.com/neetcode-gh/leetcode/pull/2114#discussion_r1083339676

Reply:
I tried to implement the same solution in the video

Reply:
Can you post link to video?

Reply:
Ahh ok just watched the video! This problem can be solved in many ways. Backtrack is a special type of DFS where you try a "path" then "undo" and try a new "path" recursively.

Take a peek at the neatcode 150 python solutions for backtrack. Most follow that particular pattern that I mentioned.

Thoughts @neetcode-gh ?